### PR TITLE
[MRG+2] Compute glyph widths similarly in Type 42 as in Type 3

### PR DESCRIPTION
**Pull request by jkseppan**
Here's my reasoning: Issue #7937 points out that text spacing is different between Type 42 and Type 3 embeddings of ttf fonts, and Type 3 looks better. One way to make them similar is to avoid the font cache and load a new `FT2Font` object, so I think the problem is that other users of these objects call `set_size` and that affects the glyph widths observed by this backend. Type 3 embeddings are not affected, and they use the `LOAD_NO_SCALE` flag, so here I change the Type 42 code to do the same and convert the sizes similarly.

In truth I have no clue how TrueType fonts work, so please review carefully.

### CODE REVIEW

**Comment by tacaswell** [comment]
guinea pig for new appveyor integration

**Comment by tacaswell** [comment]
It seems the new job got fired (https://ci.appveyor.com/project/matplotlib/matplotlib) but the CI box is still showing the link to the old one.  I assume this is just transient configuration getting crossed and the next PR will only file the new one (as I turned off the webhook to Mike's account).

File: lib/matplotlib/backends/backend_pdf.py [hunk 1]
```diff
@@ -1040,8 +1040,9 @@ def embedTTFType42(font, characters, descriptor):
             for c in characters:
                 ccode = c
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_pdf.py]: The only suggestion I have is to clean this up; `ccode` can just be the name at the beginning of the loop.

**Comment by QuLogic** [comment]
Actually, was just thinking we should probably add a test for this.

**Comment by NelleV** [comment]
:+1:  for a test

**Comment by jkseppan** [comment]
Added a test case, but it can only test that the results of the type-42 embedding stay the same as they are after this change, not that they are correct.

**Comment by NelleV** [comment]
Thanks @jkseppan !

**Comment by NelleV** [comment]
backported in v2.0.x via 9b6f7c1aa3

**Pull request merged**